### ISSUE TITLE
Ability to sign change notes with bylines

### ISSUE THREAD

**Issue by webknjaz** [python-attrs/attrs#1015]
On the wave of the recent research on crediting the contributors, I remembered one more place where this could happen (and does in some other projects) — the changelog. I've first noticed this integrated into the tox project via Sphinx's native/stdlib extlinks extension. It's proven to be nice over the last few years in several places. I've also extended the idea to link GH sponsors pages instead of GH profiles (if a sponsors page doesn't exist, GH performs a proper redirect).

The implementation is dead simple, you can see it in this line: https://github.com/aio-libs/aiohttp/blob/7bf5fa4/docs/conf.py#L163.

The usage in Towncrier change notes (suggested to be optional) looks as follows:
```rst
Made some changes -- by :user:`hynek`.
```

For preview, `Ctrl+F` entering `by @` or `@` on these pages:

- https://tox.wiki/en/latest/changelog.html
- https://docs.aiohttp.org/en/stable/changes.html
- https://setuptools.pypa.io/en/latest/history.html
- https://cheroot.cherrypy.dev/en/latest/history/

I was wondering if you wanted to give it a try in this project.

**Comment by hynek** [comment]
I’ve thought about it, but I find it too fiddly and too noisy. The metadata is all there, so that should be fine.

I think I’d be interested in a programmatic way to get all contributors of a release so I can put a summary in the release announcement. Something like “this release contains contributions by @ x, @ y, and @ z.”

I know it shouldn’t be hard, but I didn’t get around to it.

**Comment by webknjaz** [comment]
> I think I’d be interested in a programmatic way to get all contributors of a release

I've had to explore extracting this from Git a few years ago, and I recall sharing the command in somebody else's issue tracker. Let me see if I can find it again...

**Comment by webknjaz** [comment]
These could be useful, inspiration-wise:
- https://github.com/sphinx-contrib/spelling/issues/60#issuecomment-665287993
- https://github.com/ansible-community/collection_migration/pull/497/files#diff-4848cf4c7c7bc707a961bd766e376fcb92f10bedffc621ab918f419bcbbd51cfR42

TL;DR You can extract authors+committers+co-authored trailers via something like `git log '--pretty=format:%(trailers:key=Co-Authored-By,separator=%x0A)%x0A%an <%ae>%x0A%cn <%ce>' | sort | uniq` right from Git locally, with no need to access network. Tracking those back to GH accounts could be challenging, though.

**Comment by webknjaz** [comment]
> The metadata is all there, so that should be fine.

As I see it, having to navigate from the changelog to a PR or an issue (whatever is linked) and to figuring out who was involved means some extra mental overhead. Most of the time I don't care about that when I'm staring at a changelog, but it's nice to see a person who I'd be grateful to.

**Comment by hynek** [comment]
Hence my compromise with a collection of people who contributed. A changelog is still mainly to show what changed and only secondarily about who did it. Also often it's not as clear cut, when multiple people contributed etc.
